05/ContactForm.js

```diff
@@ -1,10 +1,116 @@
+import ContactFormValidation from './ContactFormValidation';
+// import account from './account/account';
 
-import account from './account';
+function ContactForm() {
+    // console.log(account);
+    const validation = new ContactFormValidation();
 
-const ContactForm = () => {
-    console.log(account);
-    return <form>??</form>;
-};
+    const init = {
+        userName: '',
+        userEmail: '',
+        userPhone: '',
+        userSubject: '',
+        userMessage: '',
+    };
+    const errors = {};
+    let errorsCount = 0;
+
+    const fields = [
+        {
+            name: 'userName',
+            type: 'text',
+            placeholder: 'Your name',
+            validationRules: { isRequired: true, regex: getNameRegex() },
+        },
+        {
+            name: 'userEmail',
+            type: 'email',
+            placeholder: 'Your email',
+            validationRules: { isRequired: true, regex: getEmailRegex() },
+        },
+        {
+            name: 'userPhone',
+            type: 'number',
+            placeholder: 'Your number',
+            validationRules: { isRequired: false, regex: getPhoneRegex() },
+        },
+        {
+            name: 'userSubject',
+            type: 'text',
+            placeholder: 'Your subject',
+            validationRules: { isRequired: true, regex: null },
+        },
+        {
+            name: 'userMessage',
+            type: 'textarea',
+            placeholder: 'Your message',
+            validationRules: { isRequired: true, regex: null },
+        },
+    ];
+
+    const reducer = (state, { name, value }) => {
+        const newState = { ...state };
+        newState[name] = value;
+        return newState;
+    };
+
+    const [state, dispatch] = useReducer(reducer, init);
+
+    const validateInputs = (e) => {
+        fields.forEach((field) => {
+            errors[field.name] = [];
+        });
+
+        fields.forEach((field) => {
+            const value = state[field.name];
+            if (field.validationRules.isRequired) {
+                if (validation.isEmpty(value)) {
+                    errors[field.name].push('Field is require');
+                }
+                if (field.validationRules.regex) {
+                    if (validation.checkDataCorrectness(field.validationRules.regex, value)) {
+                        errors[field.name].push('Incorrect format');
+                    }
+                }
+            }
+            if (!field.validationRules.isRequired) {
+                if (!validation.isEmpty(value)) {
+                    if (validation.checkDataCorrectness(field.validationRules.regex, value)) {
+                        errors[field.name].push('Incorrect format');
+                    }
+                }
+            }
+            errorsCount += errors[field.name].length;
+        });
+
+        if (errorsCount > 0) {
+            e.preventDefault();
+            console.log(errors);
+        }
```

Comment:
Ta część mogłaby być też w `ContactFormValidation` jako metoda np. `validate()`, gdzie przez parametr przekazujemy pola formularza, a zwracamy błędy w formie tablicy (pusta tablica to brak błędów).